### PR TITLE
Migrate to AndroidX, update Kotlin to 1.3.11 and Gradle to 4.10.2.

### PR DESCRIPTION
**Pull request by bernaferrari**
https://github.com/Karn/notify/issues/19

If you want to update Gradle to 5.0 there are a few deprecated things to change/remove, so I kept the most recent version that worked.

### CODE REVIEW

**Comment by Karn** [comment]
PR looks good, great first contribution! Can you run the tests locally to verify that the tests are passing?

As for gradle 5.0, well hold off for now while I go over the changes.

Cheers!

**Comment by bernaferrari** [comment]
Well, yes and no. I updated roboeletric to 4.0 and there were some fails on a few tests, probably something that changed on their api, but I'm not familiar with it, so I'm not entirely sure how to fix.
 
screen shot 2018-12-30 at 16 31 26

**Comment by Karn** [comment]
No worries, I can take a look in a few hours. In the meanwhile, try to see if the [robolectic migration guide](http://robolectric.org/migrating/#migrating-to-40) helps.

**Comment by bernaferrari** [comment]
It doesn't, the only thing it asks that you didn't do is to add "android.enableUnitTestBinaryResources=true", it removed a warning but didn't improve any test.

**Comment by Karn** [comment]
[Here](https://github.com/Karn/notify/files/2717595/test-cases-fixes.txt) is a patch file with some changes that should fix the test cases. I've removed two assertions which validate the `largeIcon` property because the comparison of `Drawables` is tedious. I will be deprecating the `largeIcon` property in a later release in favor of a property that is of type `IntRes` which will make validating this test easier and offload the resId to Bitmap conversion to the library.

You can apply the patch file by executing the following command, after which you can push the changes:

`git apply [path-to-patch-file]/test-cases-fixes.txt`

**Comment by bernaferrari** [comment]
Well.. I have no idea what you just did, but I can say I learned something before the year ended..

**Comment by bernaferrari** [comment]
Almost perfectly balanced. Thanos would be proud.
screen shot 2018-12-30 at 21 49 59

File: library/src/test/java/io/karn/notify/NotifyMetaTest.kt [hunk 1]
```diff
@@ -60,7 +61,7 @@ class NotifyMetaTest : NotifyTestBase() {
         Assert.assertEquals(testCancelOnClick, (notification.flags and NotificationCompat.FLAG_AUTO_CANCEL) != 0)
         Assert.assertEquals(testCategory, notification.category)
         Assert.assertEquals(testTimeout, notification.timeoutAfter)
-        Assert.assertEquals(1, notification.extras.getStringArray(NotificationCompat.EXTRA_PEOPLE)?.size
+        Assert.assertEquals(1, notification.extras.getStringArrayList(Notification.EXTRA_PEOPLE_LIST)?.size
```
Comment (Karn) [inline library/src/test/java/io/karn/notify/NotifyMetaTest.kt:64]: Note for later: This is the bulk of the change -- seems like API 28 has made some changes to the way that the `people` attribute is stored within the notification's `extras` Bundle.

**Review by Karn — APPROVED** [review]

**Comment by Karn** [comment]
Thanks a bunch!  🙌

**Comment by bernaferrari** [comment]
You are welcome! Also, thanks for the help. BTW, you might need to update the README (not using support anymore or kotlin 1.2). I forgot about it.

**Pull request merged**